specs/Picker-API_spec.md

```diff
@@ -0,0 +1,115 @@
+# Background
+
+The current FilePicker API for UWP has several limitations. Some of the top limitations 
```

Comment:
How closely should we be following the .NET [file open dialog](https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.openfiledialog?view=netcore-3.1)?

Reply:
Should we encapsulate the core "picking" experience (the list of folders / files) into a stand-alone control so that developers can build their own picker UI for when they need additional features? For example, Notepad customizes the standard Win32 dialog to add an "Encoding" drop-down; Photoshop adds a bunch of advanced options; etc. The control would of course only work within the constraints of whatever permissions the app had (eg, just Pictures library) but relieves the developer of having to do all the messy filesystem work and trying to copy the look and feel of the "real" dialog.

Reply:
For example, Win32 apps can embed an Explorer control or use the ancient [`DlgDirListComboBox` function](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-dlgdirlistcomboboxa).

Reply:
Could we start off with enabling the Win32 file/folder picker dialog from UWP and then make improvements to provide a modern UI.

Reply:
@ZenBird the dialog needs to run out-of-proc in the broker in order to access the files and then grant them to the UWP process. So the existing dialogs can be used, but there still needs to be some code in the middle to jump over to the broker and back again.

Reply:
Correct. V0.5 of the picker will use the existing win32 dialogs until we figure out the details of the intermediate process that will "broker" the permissions.

---
In reply to: [488307896](https://github.com/microsoft/ProjectReunion/pull/99#discussion_r488307896) [](ancestors = 488307896)